iron-component-page.html

```diff
@@ -643,8 +643,8 @@
 
         _analyzerChanged: function() {
           var analyzer = this._analyzer;
-          this._setDocElements(analyzer && analyzer.elements ? analyzer.elements : []);
-          this._setDocBehaviors(analyzer && analyzer.behaviors ? analyzer.behaviors : []);
+          this._setDocElements(analyzer ? analyzer.elementsForFolder(this.base) : []);
+          this._setDocBehaviors(analyzer ? analyzer.elementsForFolder(this.base) : []);
```

Comment:
Super cheap operation. O(N) with respect to the number of defined elements/behaviors.

```
  /**
   * Returns the elements defined in the folder containing `href`.
   * @param {string} href path to search.
   */
  elementsForFolder(href: string): ElementDescriptor[] {
    return this.elements.filter(function(element){
      return matchesDocumentFolder(element, href);
    });
  };
```